### ISSUE TITLE
How many epochs should i train?

### ISSUE THREAD

**Issue by dandelioner** [georgesterpu/Sigmedia-AVSR#16]
I have trained 2000 epochs with clean tcd-timit data by experiment_tcd_audio.py, but i can't get the result which is shown in the paper"Attention-based Audio-Visual Fusion for Robust Automatic Speech Recognition".
My experiment result:  28.27%(cer)/59.99%(wer)
Result in paper: 19.16%(cer)/45.53%(wer)

**Comment by georgesterpu** [comment]
hi @dandelioner 
Thank you for attempting to reproduce our experiments.
The default parameters in `experiment_tcd_audio.py` are not the ones that we used in the ICMI 2018 work. I will list them below for better reproducibility.

encoder_units_per_layer=((), (256, 256, 256)),
cell_type='lstm',
decoder_units_per_layer=(256, ),
optimiser='Adam',  # because we removed the support for AMSGrad due to a bug

On the Speaker Dependent (SD) partition of TCD-TIMIT I normally train the models for 400 epochs at a learning rate of 0.001 and an additional 100 epochs at 0.0001. Decaying the learning rate typically results in a substantial increase of accuracy on the test set (5-8%).

In case there is still a difference between your values and ours, I can think of two other reasons.
-The experiments for ICMI 2018 were run in last year in March, when we used an older version of the OpenFace toolkit to detect and align the faces. With OpenFace 2.0 we now get a 99% detection rate even on the more challenging LRS2 dataset. 
-Locally I was able to fix several label errors, there were many problems with the videos of volunteer 42M in particular. Unfortunately I do not have write access to the public repository of TCD-TIMIT to update the files there as well. However I don't expect to see a difference larger than 1% because of this aspect.

Please let me know if this fully answers your question.

**Comment by dandelioner** [comment]
Hi @georgesterpu 
Thank you for your reply, it's very helpful for me.
After I change the parameters, I get the result: 20.54%(cer)/44.04%(wer). It's very close to the result in paper.

**Comment by gyl1993** [comment]
I have trained 400 epochs using  learning rate 0.001 and 100 epochs using  learning rate 0.0001 with clean tcd-timit data of speaker-dependent split by experiment_tcd_av.py, but i can't get the result which is shown in the paper"Attention-based Audio-Visual Fusion for Robust Automatic Speech Recognition".
My experiment result: 30.70%(cer)/65.04%(wer)
Result in paper: 17.70%(cer)/41.90%(wer)
What should I do to reproduce the results of the paper？
